zerver/migrations/0417_backfill_remote_realmauditlog_extradata_to_json_field.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 4.0.7 on 2022-09-30 20:30
+
+import orjson
+import yaml
+from django.db import migrations
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def backfill_extra_data(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    RealmAuditLog = apps.get_model("zerver", "RealmAuditLog")
+
+    for audit_log_entry in RealmAuditLog.objects.filter(extra_data__isnull=False):
+        if audit_log_entry.event_type in (audit_log_entry.USER_FULL_NAME_CHANGED,):
+            # extra_data is a plain str, containing the old full name
+            extra_data_json = {"old_name": audit_log_entry.extra_data}
+        else:
+            try:
+                extra_data_json = orjson.loads(audit_log_entry.extra_data)
+            except orjson.JSONDecodeError:
+                # Extra data might contain improperly deserialized data generated from code like str({"foo": "bar"})
+                # that cannot be parsed by orjson. An examples of this is do_change_remote_server_plan_type.
+                # The yaml SafeLoader is capable of handling this situation and properly generates a dict
+                # for backfilling.
+                extra_data_json = yaml.safe_load(audit_log_entry.extra_data)
```

Comment:
The opposite of `str({…})` is [`ast.literal_eval`](https://docs.python.org/3/library/ast.html#ast.literal_eval), not YAML.

Reply:
I see. `literal_eval` is faster also.